include/sys/time_units.h

```diff
@@ -98,6 +98,7 @@ static TIME_CONSTEXPR ALWAYS_INLINE uint64_t z_tmcvt(uint64_t t, uint32_t from_h
 			off = rdivisor - 1U;
 		} else if (round_off) {
 			off = rdivisor / 2U;
+		} else {
 		}
```

Comment:
@andyross I'm refactoring this one but want to check on the expected behavior. The argument descriptions say we can get "round_down" behavior if we set `round_up` and `round_off` to false. In that case, we do not enter either if condition here. Does that mean "round_down" is the default  (`off = 0`)? If so, we could assert in the final else {} or we could simplify to two equal-level if statements.

We should The argument descriptions above this say

```
round_up  - Return the ceiling of the resulting fraction
round_off - Return the nearest value to the resulting fraction
                (pass both round_up/off as false to get "round_down")

Reply:
This one is a little subtle.  That function you're looking at is actually intended as something more like a macro -- those arguments are (except for the time "t" and one crazy weirdness about runtime frequency and the hpet driver) all expected to be compile time constructs which the compiler will inline to eliminate most of the code paths when expanding all the specific conversion routines.

So by construction (you can actually see the perl code I used to emit all these things in a comment in that file), "round_up" and "round_off" will never be both true.  So the "else" here is superfluous.  It's perfectly equivalent to write:


```c
		if (round_up) {
			off = rdivisor - 1U;
		}
                if (round_off) {
			off = rdivisor / 2U;
		}
```